### ISSUE TITLE
Symbol breakpoint doesn't work with C program

### ISSUE THREAD

**Issue by 0xSPIRIT** [beefytech/Beef#1874]
I'm using Beef solely as a debugger for my game written in C, which is compiled using MSVC 2022 using the command line build tools. I have a debug session that I created that simply loads the executable.

Everything works pretty well, and a lot faster than visual studio which I previously used. However, the Symbol Breakpoint functionality (Debug -> New Breakpoint -> Symbol Breakpoint) which allows you to break when the program enters a function doesn't work for me. I have to manually find the function and place a breakpoint on the line in question.

Everything else works pretty much exactly how I want it out of the box, (call stack, watch window, memory breakpoints, profiling) so if this is fixed, this would pretty much be the best debugger for my workflow!

**Comment by bfiete** [comment]
Seems to work to me. This is an unmangled C symbol?

**Comment by bfiete** [comment]
Win64?

**Comment by 0xSPIRIT** [comment]
Yes, it is Win64 and unmangled symbols since it's pure C. Is there any other information I can provide to help?

Here I've attempted to set a symbol breakpoint at the make_memory_arena function, which appears on the breakpoints list, but it doesn't seem to find it. If I manually place the breakpoint on the line itself, it works as expected.
![image](https://github.com/beefytech/Beef/assets/19749153/c5192f76-50ee-4013-a3c5-eb4cc1ea8b7a)

**Comment by 0xSPIRIT** [comment]
Testing this on a new C program it seems to work just fine, but still not with my game. I'll check my build options and see if anything is causing the issue.

**Comment by bfiete** [comment]
Oh, I see the issue right there. `make_memory_arena` is a static method, meaning it the symbol name does not get added to the global symbol table. Does Visual Studio properly set a symbol breakpoint there?

**Comment by 0xSPIRIT** [comment]
Ah, I see. Yes, Visual Studio 2022 does properly set a symbol breakpoint there.
![image](https://github.com/beefytech/Beef/assets/19749153/ac939111-6a9b-44b0-a803-f51377fd09c7)
